ciphey/basemods/Crackers/affine.py

```diff
@@ -0,0 +1,125 @@
+    """
+
+    def getInfo(self, ctext: str) -> CrackInfo:
+        return CrackInfo(
+            success_likelihood=0.1,
+            success_runtime=1e-5,
+            failure_runtime=1e-5,
+        )
+
+    @staticmethod
+    def getTarget() -> str:
+        return "Affine"
+
+    def attemptCrack(self, ctext: str) -> List[CrackResult]:
+        """
+        Brute forces all the possible combinations of a and b to attempt to crack the cipher.
+        """
+        logger.trace("Attempting Affine brute force.")
+        candidates = []
+
+        # a and b are coprime if gcd(a,b) is 1.
+        possible_a = [
+            a
+            for a in range(1, self.ALPHABET_LENGTH)
+            if mathsHelper.gcd(a, self.ALPHABET_LENGTH) == 1
+        ]
+        logger.debug(
+            f"Trying Affine Cracker with {len(possible_a)} a-values and {self.ALPHABET_LENGTH} b-values"
+        )
+
+        for a in possible_a:
+            a_inv = mathsHelper.mod_inv(a, self.ALPHABET_LENGTH)
+            # If there is no inverse, we cannot decrypt the text
+            if a_inv is None:
+                continue
+            for b in range(self.ALPHABET_LENGTH):
+                # Pass in lowered text. This means that we expect alfabets to not contain both 'a' and 'A'.
+                translated = self.decrypt(ctext.lower(), a_inv, b, self.ALPHABET_LENGTH)
+
+                candidate_probability = self.plaintext_probability(translated)
+                if candidate_probability > self.PLAINTEXT_PROB_THRESHOLD:
+                    candidates.append(
+                        CrackResult(
+                            value=fix_case(translated, ctext), key_info=f"a={a}, b={b}"
+                        )
+                    )
+        logger.debug(f"Affine Cipher returned {len(candidates)} candidates")
+        return candidates
+
+    def plaintext_probability(self, translated: str) -> float:
+        """
+        Analyses the translated text and applies the chi squared test to see if it is a probable plaintext candidate
+        Returns the probability of the chi-squared test.
+        """
+        analysis = cipheycore.analyse_string(translated)
+        return cipheycore.chisq_test(analysis, self.expected)
+
+    def decrypt(self, text: str, a_inv: int, b: int, m: int) -> str:
+        """
+        Each letter is decrypted at D(x) = a_inv (x - b) mod m where x is the char
+        We treat the char value as its index in the alphabet, so if
+        the alphabet is 'abcd....' and the char is 'b', it has the value 1.
+        """
+        return "".join([self.decryptChar(char, a_inv, b, m) for char in text])
+
+    def decryptChar(self, char: str, a_inv: int, b: int, m: int) -> str:
+
+        # We lower the alphabet since both ctext and alphabet need to be in the same case in order
+        # to perform the shifts. The translated text will have fixed case after the translation anyways.
+        alphabet = [x.lower() for x in self.group]
```

Comment:
What is this variable used for? You assign to it, but it never gets used anywhere.